src/Table/index.jsx

```diff
@@ -73,14 +146,28 @@ Table.propTypes = {
       PropTypes.string,
       PropTypes.element,
     ]).isRequired,
+    columnSortable: isRequiredIf(PropTypes.bool, props => props.tableSortable),
+    onSort: isRequiredIf(PropTypes.func, props => props.columnSortable),
   })).isRequired,
   headingClassName: PropTypes.arrayOf(PropTypes.string),
+  tableSortable: PropTypes.bool,
+  /* eslint-disable react/require-default-props */
+  defaultSortedColumn: isRequiredIf(PropTypes.string, props => props.tableSortable),
+  /* eslint-disable react/require-default-props */
+  defaultSortDirection: isRequiredIf(PropTypes.string, props => props.tableSortable),
+  sortButtonsScreenReaderText: isRequiredIf(PropTypes.object, props => props.columnSortable),
```

Comment:
Does `isRequiredIf` not allow you to specify a shape? It'd be good to be explicit about the keys required in this object.